qiskit/pulse/library/symbolic_pulses.py

```diff
@@ -0,0 +1,873 @@
+                if p.name == "t":
+                    # Argument "t" must be placed at first. This is a vector.
+                    params.insert(0, p)
+                    continue
+                params.append(p)
+
+            if _optional.HAS_SYMENGINE:
+                try:
+                    lamb = sym.lambdify(params, [value], real=False)
+
+                    def _wrapped_lamb(*args):
+                        if isinstance(args[0], np.ndarray):
+                            # When the args[0] is a vector ("t"), tile other arguments args[1:]
+                            # to prevent evaluation from looping over each element in t.
+                            t = args[0]
+                            args = np.hstack(
+                                (
+                                    t.reshape(t.size, 1),
+                                    np.tile(args[1:], t.size).reshape(t.size, len(args) - 1),
+                                )
+                            )
+                        return lamb(args)
+
+                    func = _wrapped_lamb
+                except RuntimeError:
+                    # Currently symengine doesn't support complex_double version for
+                    # several functions such as comparison operator and piecewise.
+                    # If expression contains these function, it fall back to sympy lambdify.
+                    # See https://github.com/symengine/symengine.py/issues/406 for details.
+                    import sympy
+
+                    func = sympy.lambdify(params, value)
+            else:
+                func = sym.lambdify(params, value)
+
+            self.lambda_funcs[key] = func
+
+
+class SymbolicPulse(Pulse):
+    r"""The pulse representation model with parameters and symbolic expressions.
+
+    A symbolic pulse instance can be defined with an envelope and parameter constraints.
+    Envelope and parameter constraints should be provided as symbolic expressions.
+    Rather than creating a subclass, different pulse shapes can be distinguished by
+    the instance attributes :attr:`SymbolicPulse.envelope` and :attr:`SymbolicPulse.pulse_type`.
+
+    The symbolic expressions must be defined either with SymPy_ or Symengine_.
+    Usually Symengine-based expression is much more performant for instantiation
+    of the :class:`SymbolicPulse`, however, it doesn't support every functions available in SymPy.
+    You may need to choose proper library depending on how you define your pulses.
+    Symengine works in the most envelopes and constraints, and thus it is recommended to use
+    this library especially when your program contains a lot of pulses.
+    Also note that Symengine has the limited platform support, in particular, for 32-bit platforms
+    such as ``i686`` and ``arm``, or Ubuntu Server such as ``s390x``. You must confirm Symengine
+    is available in your environment, otherwise it must fall back into SymPy.
```

Comment:
I think the platform support needs a small tweak:

```suggestion
    Also note that Symengine has the limited platform support and may not be available for your local system.
    Symengine is a required dependency for Qiskit on platforms that support it will always be installed along
    with Qiskit on macOS ``x86_64`` and ``arm64``, and Linux ``x86_64``, ``aarch64``, and ``ppc64le``. For 64bit
    Windows users they will need to manual install it. For 32-bit platforms such as ``i686`` and ``armv7`` Linux,
    and on Linux ``s390x`` there are no pre-compiled packages available and to use symengine you'll need to
    compile it from source. If Symengine is not available in your environment SymPy will be used.
```